### ISSUE TITLE
Tags support

### ISSUE THREAD

**Issue by tj** [apex/up#125]
Support for tagging so it's easier to break down billing etc

**Comment by rjsamson** [comment]
Is this still under consideration? We use tags for breaking everything down billing-wise in our aws environments, and it's the one real blocker to us being able to use up.

**Comment by tj** [comment]
Yep! It's not super high priority but definitely on the list. Is there anything specific you'd need, or would just tagging with the app name by default be ok?

**Comment by rjsamson** [comment]
Ideally we'd prefer the ability to add arbitrary tags in k/v pairings - we use a couple of custom tags besides Name to identify and route billing for applications (like ci or a cmdb identifer), and beyond that we use these tags to drive role based aws permissions on a per-app basis, so the ability to add custom tag names and values to everything in the stack would be really great. Would that be possible?

**Comment by tj** [comment]
For #453 I was thinking optional command(s) could basically transform the cloudformation as they need via stdin and just stdout the results, so that could work. 

As far as out-of-the-box support I'm not sure I'd want `tags: {}` all over the json—seems a bit overkill for most people—but I could see having one root-level `tags`.

**Comment by rjsamson** [comment]
Yeah - one root level `tags` would work great for us.

**Comment by tj** [comment]
Noted! That sounds reasonable to me.
